templates/home.html

```diff
@@ -8,66 +9,78 @@
         }
     </style>
 {% endblock %}
+
 {% block pagecontent %}
-    <br>
     {% include "toast.html" %}
-    <h1>{{ session.first_name }} {{ session.name }}</h1>
-    <hr>
-    {% if is_researcher %}
-        <h2 class="sub-header">Course(s) assigned this year ({{ dynamic_year }} - {{ dynamic_year + 1 }})</h2>
-        <table class="table table-hover">
-            <thead>
-            <tr>
-                <th>Code</th>
-                <th>Title</th>
-            </tr>
-            </thead>
-            <tbody>
-            <tr>
-                <td colspan="2">No courses assigned this year.</td>
-            </tr>
-            </tbody>
-        </table>
-        <div class="container-fluid">
-            <div class="card">
+
+    <div class="container-fluid">
+        <!-- Header Section -->
+        <h1 class="mt-3 mb-3">{{ session.first_name }} {{ session.name }}</h1>
+        <hr>
+
+        <!-- Researcher Section -->
+        {% if is_researcher %}
```

Comment:
Maybe do role checks in a more harmonized way. You are checking if the user is an admin through session, if they are a teacher through a User result from the DB and if they are a researcher like this. How about storing the role in session @anthonygego ?

Also, where is `is_researcher` defined ?

Reply:
Its defined in the context_processor

Reply:
Thanks, I will have a look